### PR TITLE
Wallet

### PR DESCRIPTION
**Pull request by janbormet**

### CODE REVIEW

File: wallet/backend.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+func init() {
+	wallet.SetBackend(Backend)
+}
+
+func (b backend) NewAddress() wallet.Address {
+	return &address.Address{}
+}
+
+func (b backend) DecodeSig(reader io.Reader) (wallet.Sig, error) {
```
Comment (ndzik) [inline wallet/backend.go:25]: Add a docstring reinforcing that we only use padded signatures with this backend. The interface method itself is too ambiguous for our use-case and someone having to adapt this backend might miss out on this fact.

File: encoding/signature.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package encoding
+
+import (
+	"github.com/nervosnetwork/ckb-sdk-go/v2/types"
+	"github.com/nervosnetwork/ckb-sdk-go/v2/types/molecule"
+	"perun.network/perun-ckb-backend/wallet"
+)
+
+func NewDEREncodedSignatureFromPadded(paddedSignature []byte) (*molecule.Bytes, error) {
```
Comment (ndzik) [inline encoding/signature.go:9]: Unused?

File: wallet/signature.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		return nil, fmt.Errorf("signature is of wrong length. Expected %d bytes, got %d bytes", PaddedSignatureLength, len(sig))
+	}
+	for i := len(sig) - 1; i >= 0; i-- {
+		if sig[i] == MarkerByte {
+			return sig[:i], nil
+		}
+		if sig[i] != ZeroByte {
+			return nil, fmt.Errorf("invalid padding")
+		}
+	}
```
[the remark this answers is not in the document]
Comment (janbormet) [reply wallet/signature.go:51]: We discussed this, I adjusted the docstring.

File: encoding/signature.go (hunk 2 above)
Comment (janbormet) [reply encoding/signature.go:9]: yes

File: wallet/backend.go (hunk 1 above)
Comment (janbormet) [reply wallet/backend.go:25]: Done!

File: wallet/address/address.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+)
+
+const (
+	UncompressedPublicKeyLength = 65
+	CompressedPublicKeyLength   = 33
+	Uint64Length                = 8
+	SerializedAddressLength     = CompressedPublicKeyLength + types.HashLength + types.HashLength + Uint64Length
+)
+
+type Address struct {
```
[the remark this answers is not in the document]
Comment (janbormet) [reply wallet/address/address.go:21]: done

**Review by ndzik — APPROVED** [review]
LGTM

**Pull request merged**